### ISSUE TITLE
Reduce/manage the number of settings in settings.js

### ISSUE THREAD

**Issue by sbc100** [emscripten-core/emscripten#8257]
tldr; I'd like to try to reduce the number of user-visible settings in settings.js

There are currently over 230 settings values in settings.js.   

These settings service many difference purposes.  Some are internal-only settings for communicating between python and js code.   Some effect the generated code.  Some effect library code.

All these settings a modifiable by the user via command line and visible to the user in via the JS "pre-processor".  (Are they visible in any other way other than this ifdef system in pre-processor)?

Many of these settings can take more than just 2 values so that combinatorial effect of this is enormous.   There is simply no way we can test even a tiny subset of this space.

Some immediate things we can do:

1. Remove any orphaned settings (settings that are no longer used anywhere)
2. Find a way to distinguish between user-visible settings and non-user-visible settings
   - Which settings can come from the command line?
   - Which settings are internal-only?
   - Which setttings are visible to user-JS code?
2. Raise that bar for adding new settings.  At least user-visible ones.
3. Look for opportunities to retire settings.

We've already kind of been doing this but I wanted to create an issue to track this.

**Comment by jgravelle-google** [comment]
This is overall a big +1 from me.

I believe non-user-visible settings are still visible via the command line though, because `emcc` can call sub-invocations of `emcc` with different, internal-only arguments. (e.g., here https://github.com/emscripten-core/emscripten/blob/883905425c5cc988374f728eeccc11d5c1688c49/tools/gen_struct_info.py#L396)
I'd like to say we should disallow those settings via command-line, but then we'd need to have some other mechanism for forwarding that information.
A simple immediate solution might be to move them to an `internal_settings.js` file, and for those arguments change `-s` into `-s-internal`.

Another observation is that because the emscripten driver should be the only user of these internal flags, whatever mechanism we use can be more verbose than if it were user-facing.

**Comment by sbc100** [comment]
I feel like we i've done a pretty good job on this.   Lets call it closed for now/

**Comment by sbc100** [comment]
Sorry "we've" done a pretty good job
